spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractLocalDeployerSupport.java

```diff
@@ -73,16 +78,22 @@ final protected LocalDeployerProperties getLocalDeployerProperties() {
 	 * @param request the request for the application to execute
 	 * @return the build command as a string array
 	 */
-	private String[] buildExecutionCommand(AppDeploymentRequest request) {
+	private String[] buildExecutionCommand(AppDeploymentRequest request, Map<String, String> args) {
 		ArrayList<String> commands = new ArrayList<String>();
-		commands.add(properties.getJavaCmd());
 		Map<String, String> deploymentProperties = request.getDeploymentProperties();
 		ExecutionCommandBuilder commandBuilder = new ExecutionCommandBuilder();
-		// Add Java System Properties (ie -Dmy.prop=val) before main class or -jar
-		commandBuilder.addJavaOptions(commands, deploymentProperties, properties);
-		commandBuilder.addJavaExecutionOptions(commands, request);
-		commands.addAll(request.getCommandlineArguments());
-		logger.debug("Java Commands = " + commands);
+		if (request.getResource() instanceof DockerResource) {
+			commandBuilder.addDockerOptions(commands, request, args);
```

Comment:
Ah, sorry, did not realize this was modified by side effect. Given the method is currently `void`, maybe it could be returned instead? (along with my other comment about making that class a Strategy as well)